client/js/store/statement/Statement.js

```diff
@@ -123,67 +160,47 @@ function setUpdatedProps (data) {
   statement.fragmentsTotal = statement.fragmentsCount || 0
-  statement.files = statement.files || []
-  statement.attachments = statement.attachments || []
-  statement.sourceAttachment = statement.sourceAttachment || ''
+  statement.genericAttachments = statement.genericAttachments || []
+  statement.id = el.id
   statement.initialFilteredFragmentsCount = statement.filteredFragmentsCount || 0 // This Information is missing from BE-Side by now
-  statement.phase = Translator.trans(statement.phase)
   statement.isFiltered = meta.isFiltered || false
-  statement.orgaName = statement.organisationName
   statement.orgaDepartmentName = statement.organisationDepartmentName
+  statement.orgaName = statement.organisationName
+  statement.phase = Translator.trans(statement.phase)
+  statement.sourceAttachment = statement.sourceAttachment || ''
 
   if (hasOwnProp(el, 'relationships')) {
     const relationships = Object.keys(el.relationships)
 
     // Get the data for the relationships and put it into the statement-element
     relationships.forEach(relationKey => {
       const relation = el.relationships[relationKey]
+      const data = relation?.data
+
       // For 1-n Relations
-      if (relation.data instanceof Array) {
-        if (relation.data.length > 0) {
-          const ids = relation.data.map(id => id.id)
-          const type = relation.data[0].type
-
-          statement[relationKey] = includes.filter(incl => ids.includes(incl.id) && type === incl.type)
-          statement[relationKey] = statement[relationKey].map(statementRel => Object.assign(statementRel.attributes, { id: statementRel.id }))
-
-          if (type === 'SourceStatementAttachment' && hasOwnProp(statement[relationKey][0], 'id')) {
-            const attachment = includes
-              .filter(incl => incl.type === 'SourceStatementAttachment')
-              .filter(incl => statement[relationKey][0].id === incl.id)
-
-            if (hasOwnProp(attachment[0], 'relationships')) {
-              const sourceAttachment = includes
-                .filter(incl => incl.type === 'File')
-                .filter(incl => attachment[0].relationships.file.data.id === incl.id)
-                .map(sourceAtt => Object.assign(sourceAtt.attributes, { id: sourceAtt.id }))
-
-              statement.sourceAttachment = sourceAttachment[0]
-            } else {
-              statement.sourceAttachment = undefined
-            }
-          }
-        } else {
-          statement[relationKey] = []
+      if (Array.isArray(data)) {
+        const items = data.length ? mapRelations(data, includes) : []
+        statement[relationKey] = items
+
+        if (relationKey === 'genericAttachments') {
+          statement.genericAttachments = extractFileAttachments(items, includes)
+        }
+
+        if (relationKey === 'sourceAttachment') {
+          statement.sourceAttachment = extractFileAttachments(items, includes, { single: true })
```

Comment:
couldn't we "optimize away" the single flag if we expect an array here like:
```suggestion
          statement.sourceAttachment = extractFileAttachments(items, includes)[0] ?? null
```

Reply:
ok, got you, agree https://github.com/demos-europe/demosplan-core/pull/4836/commits/1b38aa15d570a3a03bf93f0c114bcea551d884c6